wgpu-types/src/lib.rs

```diff
@@ -361,30 +434,49 @@ pub fn read_spirv<R: io::Read + io::Seek>(mut x: R) -> io::Result<Vec<u32>> {
 }
 
 bitflags::bitflags! {
+    /// Describes the shader stages that this binding will be visible from.
+    ///
+    /// These can be combined so something that is visible from both vertex and fragment shaders can be defined as:
+    ///
+    /// `ShaderStage::VERTEX | ShaderStage::FRAGMENT`
     #[repr(transparent)]
     #[cfg_attr(feature = "trace", derive(Serialize))]
     #[cfg_attr(feature = "replay", derive(Deserialize))]
     pub struct ShaderStage: u32 {
+        /// Binding is not visible from any shader stage
         const NONE = 0;
+        /// Binding is visible from the vertex shader of a render pipeline
         const VERTEX = 1;
+        /// Binding is visible from the fragment shader of a render pipeline
         const FRAGMENT = 2;
+        /// Binding is visible from the compute shader of a compute pipeline
         const COMPUTE = 4;
     }
 }
 
+/// Dimensions of a particular texture view.
 #[repr(C)]
 #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
 #[cfg_attr(feature = "trace", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 pub enum TextureViewDimension {
+    /// A one dimensional texture. `texture1D` in glsl shaders.
     D1,
+    /// A two dimensional texture. `texture2D` in glsl shaders.
     D2,
+    /// A two dimensional array texture. `texture2DArray` in glsl shaders.
     D2Array,
+    /// A cubemap texture. `textureCube` in glsl shaders.
     Cube,
+    /// A cubemap array texture. `textureCubeArray` in glsl shaders.
     CubeArray,
+    /// A three dimensional texture. `texture3D` in glsl shaders.
     D3,
 }
 
+/// Alpha blend factor.
+///
+/// Alpha blending is very complicated: see the opengl or vulkan spec for more information.
```

Comment:
"see the OpenGL or Vulkan spec, or MSDN docs for D3D, for more information"

If you don't want to mention The Best API, at least capitalize OpenGL and Vulkan properly smh